From 35fc80061d1c71cf4496321377279c2be63dba38 Mon Sep 17 00:00:00 2001
From: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Date: Thu, 13 Dec 2018 15:08:46 +0700
Subject: [PATCH 716/909] iommu/ipmmu-vmsa: Correct whitelist devices
 assignment

commit a5266d298124874c2c06b8b13d073f6ecc2ee355 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch corrects whitelist devices assignment, in particular:
- Remove unavailable USB-DMAC ch2/3 devices on r8a7796 (R-Car M3) whitelist
- Add missing USB2.0 Host ch1 device on r8a77965 (R-Car M3N) whitelist
- Fix typo for Audio DMAC ch26 device

Signed-off-by: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/iommu/ipmmu-vmsa-whitelist.h | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/iommu/ipmmu-vmsa-whitelist.h b/drivers/iommu/ipmmu-vmsa-whitelist.h
index 8e5368a83ee2..4a846d6991b5 100644
--- a/drivers/iommu/ipmmu-vmsa-whitelist.h
+++ b/drivers/iommu/ipmmu-vmsa-whitelist.h
@@ -156,7 +156,7 @@
 				 M_AUDIO_DMAC_20 | M_AUDIO_DMAC_21 | \
 				 M_AUDIO_DMAC_22 | M_AUDIO_DMAC_23 | \
 				 M_AUDIO_DMAC_24 | M_AUDIO_DMAC_25 | \
-				 M_AUDIO_DMAC_16 | M_AUDIO_DMAC_27 | \
+				 M_AUDIO_DMAC_26 | M_AUDIO_DMAC_27 | \
 				 M_AUDIO_DMAC_28 | M_AUDIO_DMAC_29 | \
 				 M_AUDIO_DMAC_30 | M_AUDIO_DMAC_31)
 #define H3_IPMMU_DS0_MASTER	(M_SYS_DMAC_0 | M_SYS_DMAC_1 | \
@@ -191,8 +191,7 @@
 				 M_FCPVD_1 | M_FCPVD_2 | M_FCPVD_3)
 #define M3_IPMMU_HC_MASTER	(M_PCIE_0 | M_PCIE_1 | M_USB2H_0 | \
 				 M_USB2H_1 | M_USB_DMAC_0 | M_USB_DMAC_1 | \
-				 M_USB3H_0 | M_USB3F_0 | M_USB_DMAC_2 | \
-				 M_USB_DMAC_3)
+				 M_USB3H_0 | M_USB3F_0)
 #define M3_IPMMU_MP_MASTER	(M_AUDIO_DMAC_0 | M_AUDIO_DMAC_1 | \
 				 M_AUDIO_DMAC_2 | M_AUDIO_DMAC_3 | \
 				 M_AUDIO_DMAC_4 | M_AUDIO_DMAC_5 | \
@@ -206,7 +205,7 @@
 				 M_AUDIO_DMAC_20 | M_AUDIO_DMAC_21 | \
 				 M_AUDIO_DMAC_22 | M_AUDIO_DMAC_23 | \
 				 M_AUDIO_DMAC_24 | M_AUDIO_DMAC_25 | \
-				 M_AUDIO_DMAC_16 | M_AUDIO_DMAC_27 | \
+				 M_AUDIO_DMAC_26 | M_AUDIO_DMAC_27 | \
 				 M_AUDIO_DMAC_28 | M_AUDIO_DMAC_29 | \
 				 M_AUDIO_DMAC_30 | M_AUDIO_DMAC_31)
 #define M3_IPMMU_DS0_MASTER	(M_SYS_DMAC_0 | M_SYS_DMAC_1 | \
@@ -239,8 +238,8 @@
 /* Support masters for r8a77965 */
 #define M3N_IPMMU_VI0_MASTER	(M_VIN_0_3 | M_VIN_4_7 | M_FCPVD_0 | M_FCPVD_1)
 #define M3N_IPMMU_HC_MASTER	(M_PCIE_0 | M_PCIE_1 | M_SATA | M_USB2H_0 | \
-				 M_USB_DMAC_0 | M_USB_DMAC_1 | M_USB3H_0 | \
-				 M_USB3F_0)
+				 M_USB2H_1 | M_USB_DMAC_0 | M_USB_DMAC_1 | \
+				 M_USB3H_0 | M_USB3F_0)
 #define M3N_IPMMU_MP_MASTER	(M_AUDIO_DMAC_0 | M_AUDIO_DMAC_1 | \
 				 M_AUDIO_DMAC_2 | M_AUDIO_DMAC_3 | \
 				 M_AUDIO_DMAC_4 | M_AUDIO_DMAC_5 | \
@@ -254,7 +253,7 @@
 				 M_AUDIO_DMAC_20 | M_AUDIO_DMAC_21 | \
 				 M_AUDIO_DMAC_22 | M_AUDIO_DMAC_23 | \
 				 M_AUDIO_DMAC_24 | M_AUDIO_DMAC_25 | \
-				 M_AUDIO_DMAC_16 | M_AUDIO_DMAC_27 | \
+				 M_AUDIO_DMAC_26 | M_AUDIO_DMAC_27 | \
 				 M_AUDIO_DMAC_28 | M_AUDIO_DMAC_29 | \
 				 M_AUDIO_DMAC_30 | M_AUDIO_DMAC_31)
 #define M3N_IPMMU_DS0_MASTER	(M_SYS_DMAC_0 | M_SYS_DMAC_1 | \
-- 
2.17.1

